### PR TITLE
wfe: allow POST-as-GET to dir. and newNonce endpoints

### PR DESCRIPTION
**Pull request by cpu**
[RFC 8555 §6.3](https://tools.ietf.org/html/rfc8555#section-6.3) says the server's `directory` and `newNonce` endpoints should support POST-as-GET as well as GET.

This PR adopts a fix from eggsampler's [Pebble fork](eggsampler@ce1b722). Thanks @eggsampler !

Along the way to avoid the ire of the linter I replaced the numerous occurrences of "GET", "POST", and "HEAD" with the correct exported `http` package constants.

Resolves https://github.com/letsencrypt/pebble/issues/291

### CODE REVIEW

File: wfe/wfe.go [hunk 1]
```diff
@@ -431,23 +431,23 @@ func (wfe *WebFrontEndImpl) handleCertStatusBySerial(
 
 func (wfe *WebFrontEndImpl) Handler() http.Handler {
 	m := http.NewServeMux()
-	// GET only handlers
-	wfe.HandleFunc(m, DirectoryPath, wfe.Directory, "GET")
-	// Note for noncePath: "GET" also implies "HEAD"
-	wfe.HandleFunc(m, noncePath, wfe.Nonce, "GET")
+	// GET & POST handlers
+	wfe.HandleFunc(m, DirectoryPath, wfe.Directory, http.MethodGet, http.MethodPost)
```
Comment (jsha) [inline wfe/wfe.go]: It looks like all the HandleFuncs now allow MethodPost. Maybe it's better to change the HandleFunc definition so it automatically adds POST? And then adds whatever methods are left in the varargs?

File: wfe/wfe.go [hunk 2]
```diff
@@ -478,6 +478,20 @@ func (wfe *WebFrontEndImpl) Directory(
 		"keyChange":  keyRolloverPath,
 	}
 
+	// RFC 8555 §6.3 says the server's directory endpoint should support
```
Comment (jsha) [inline wfe/wfe.go]: I like the proper section markers. Hooray for broad Unicode support! :-D

File: wfe/wfe.go (hunk 1 above)
Comment (cpu) [reply wfe/wfe.go]: I feel like that will make reading the code more confusing and we're unlikely to be adding new handlers with much frequency. Do you feel strongly about it? I have a light preference for the explicit version.

**Comment by cpu** [comment]
> there's no reason to switch over to POST-as-GET for subsequent runs.

Agreed, but the spec says what it says. All of the POST-as-GET stuff is jank and this is just another weird artifact of changing the spec with minimal input at the very end of the process. ¯\\\_(ツ)\_/¯

File: wfe/wfe.go (hunk 1 above)
Comment (jsha) [reply wfe/wfe.go]: I don't feel strongly about it; the explicit version is fine.

**Pull request merged**